### ISSUE TITLE
Expand basic.feature to cover elementary usage of Minishift

### ISSUE THREAD

**Issue by agajdosi** [minishift/minishift#548]
Currently basic.feature does not cover much of Minishift use and needs an expansion. After short call on last Friday and further discussion with @amitkrout on this topic we propose following for basic.feature:

- As we are at the start with integration tests, we should focus on covering developers' use first, meaning we should prioritize time over case coverage.
- It would be good to cover elementary usage of Minishift in basic.feature, ideally following simple user story,
- which can be further expanded to cover more cases if those are not time consuming (no extra Minishift start, no extra creations of OpenShift applications).
- This will make basic.feature a relatively fast test for frequent use of developers, which will assure them that Minishift is passing smoke test and at least basic sanity on given platform.

Story we have in mind is: one wants to try OpenShift, for this case she choose Minishift, she should be able to start Minishift, she should be able to do ssh into machine, be able to deploy example application on OpenShift via oc command, able to stop and delete Minishift. These steps are time consuming but are needed. Further we can add less consuming cases to cover more. As result of this we are currently proposing following new scenarios aka tests:

- check that OpenShift console is accessible, that OpenShift returns at /healthz and /healthz/ready "ok",
- check that oc is on its place,
- make sure an example application can be deployed on OpenShift - this is the only new feature which is really time expensive, could take from 3 to 5 mins (will be covered in #750)
- delete application is fine,
- check that `minishift config set / get / view / unset` works - we would not be able to test real functionality of these changes as this would cost one extra start, but we can at least check that config is w/r-able, that commands works well,
- `minishift openshift config set / view` - probably no need to test `minishift openshift restart` as this functionality is part of `set`,
- check that external network is accessible from both host and guest,
- would be nice to check that `minishift status` is actually working.

WDYT? Is there anything which should be added?

**Comment by hferentschik** [comment]
> As we are at the start with integration tests, we should focus on covering developers' use first, meaning we should prioritize time over case coverage.

+1

> check that OpenShift console is accessible, that OpenShift returns at /healthz and /healthz/ready "ok",

Here we probably want another common helper which allows us to probe an URL and expect a given HTTP status and potentially response content

> check that oc is on its place

+1

> make sure an example application can be deployed on OpenShift - this is the only new feature which is really time expensive, could take from 3 to 5 mins

+1, we need to decide on how we want to handle determining on when the app is ready

> check that minishift config set / get / view / unset works - we would not be able to test real functionality of these changes as this would cost one extra start, but we can at least check that config is w/r-able, that commands works well,

+1 One can start with just checking that the commands set the right values in the underlying JSON config. Whether the set values are properly applied then, could be deferred. To do this, one does not even have to start Minishift.

> check that external network is accessible from both host and guest,

hmm, not sure about this one. How is this something that Minishift controls. Why is this something we need/should test?

> would be nice to check that minishift status is actually working.

+1 That's a must almost. Do you see any issues with that?

What's about 'minishift ip' returning the expected IP. 'minishift console --url' returning the URL to the console, 'minishift logs' printing some log lines

**Comment by agajdosi** [comment]
@hferentschik Thank you for feedback.

> Here we probably want another common helper which allows us to probe an URL and expect a given HTTP status and potentially response content

Currently I have function written for OpenShift console, but I was thinking the same: moving the part which retrieves url of console into separate function which will then call more common URL retriever. Then we will be able to add functions for testing application url or similar. Will go on that today.

> One can start with just checking that the commands set the right values in the underlying JSON config.

With this feature I am done with get, set and unset commands, for view I am waiting for Praveen's PR. With config file I did a small mistake that I am checking text in file, not using JSON decoder.

> +1 That's a must almost. Do you see any issues with that?

I do not remember any.
 >What's about 'minishift ip' returning the expected IP. 'minishift console --url' returning the URL to the console, 'minishift logs' printing some log lines

Adding this into my plan.

Also I started to support negation in my new scenarios, so for example:
```
Then config file contains '"<property>": <value>'
[...]
Then config file does not contain '"<property>": <value>'
```
^ will be reworked to not use text but instead check for JSON, but logic will be the same.

**Comment by agajdosi** [comment]
Part of this issue focused on testing usage of OpenShift was moved into separate issue #750.

**Comment by hferentschik** [comment]
Merged via pull request #791
